dev/ProgressRing/ProgressRing.cpp

```diff
@@ -41,18 +47,70 @@ void ProgressRing::OnRangeBasePropertyChanged(const winrt::DependencyObject&, co
     UpdateSegment();
 }
 
+void ProgressRing::OnForegroundPropertyChanged(const winrt::DependencyObject&, const winrt::DependencyProperty&)
+{
+    ApplyLottieAnimation();
+}
+
+void ProgressRing::OnBackgroundPropertyChanged(const winrt::DependencyObject&, const winrt::DependencyProperty&)
+{
+    ApplyLottieAnimation();
+}
+
 void ProgressRing::OnStrokeThicknessPropertyChanged(const winrt::DependencyPropertyChangedEventArgs& args)
 {
+    ApplyLottieAnimation();
     UpdateRing();
 }
 
-winrt::Size ProgressRing::ComputeEllipseSize(double thickness, double actualWidth, double actualHeight)
+void ProgressRing::OnIsIndeterminatePropertyChanged(const winrt::DependencyPropertyChangedEventArgs& args)
+{
+    UpdateStates();
+}
+
+void ProgressRing::ApplyLottieAnimation()
+{
+    if (auto&& player = m_player.get())
+    {
+        // ProgressRing only accounts for ActualWidth to ensure that it is always a circle
+        const float diameter = static_cast<float>(ActualWidth());
+        const auto size = winrt::Size({ diameter, diameter });
+        const auto foreground = Foreground().try_as<winrt::SolidColorBrush>().Color();
+        const auto background = Background().try_as<winrt::SolidColorBrush>().Color();
+        player.Source(winrt::make<ProgressRingLoading>(StrokeThickness(), size, foreground, background));
+    }
+}
+
+void ProgressRing::UpdateStates()
+{
+    if (IsIndeterminate())
+    {
+        winrt::VisualStateManager::GoToState(*this, s_IndeterminateStateName, true);
+
+        if (auto&& player = m_player.get())
+        {
+            player.PlayAsync(0, 1, true);
+        }
+    }
+    else if (!IsIndeterminate())
+    {
+        winrt::VisualStateManager::GoToState(*this, s_DeterminateStateName, true);
+
+        if (auto&& player = m_player.get())
+        {
+            player.Stop();
+        }
+    }
+}
+
+winrt::Size ProgressRing::ComputeCircleSize(double thickness, double actualWidth)
 {
     const double safeThickness = std::max(thickness, static_cast<double>(0.0));
-    const double width = std::max((actualWidth - safeThickness) / 2.0, 0.0);
-    const double height = std::max((actualHeight - safeThickness) / 2.0, 0.0);
 
-    return {static_cast<float>(width), static_cast<float>(height)};
+    // ProgressRing only accounts for ActualWidth to ensure that it is always a circle
+    const double diameter = std::max((actualWidth - safeThickness) / 2.0, 0.0);
```

Comment:
>diameter [](start = 17, length = 8)

This is the radius, not the diameter.

Reply:
I'm not too sure of the calculations that goes on here as it was referenced from the community toolkit, but I think you're right that it should be renamed as `radius`

Reply:
I think that the method name is wrong two, you aren't returning the circle's size.

---
In reply to: [369290852](https://github.com/microsoft/microsoft-ui-xaml/pull/1858#discussion_r369290852) [](ancestors = 369290852)

Reply:
Looking closer I think you could just return a double that is the radius here. Since where you use this helper you don't actually need a size.

---
In reply to: [369730525](https://github.com/microsoft/microsoft-ui-xaml/pull/1858#discussion_r369730525) [](ancestors = 369730525,369290852)